File: 006/Erlang/problem006.erl
-module(problem006).
-export([diff_squares_sums/1, sum_of_squares/1, square_of_sum/1]).

diff_squares_sums(N) -> square_of_sum(N) - sum_of_squares(N).

sum_of_squares(N) -> N * (N + 1) * (2 * N + 1) div 6.

square_of_sum(N) -> Sum = N * (N + 1) div 2, Sum * Sum.

File: 006/Python/problem006.py
def diff_squares_sums(n):
    return square_of_sum(n) - sum_of_squares(n)
    
def sum_of_squares(n):
    return n * (n + 1) * (2 * n + 1) / 6
    
def square_of_sum(n):
    sum = n * (n + 1) / 2
    return sum * sum
    
